ci_scripts/test_sync_target_db.py

```diff
@@ -0,0 +1,270 @@
+import json
+import os
+
+from collections import namedtuple
+from unittest import TestCase, mock
+
+from ci_scripts import sync_target_database
+
+ONLINE_TARGET_DATABASE = """[
```

Comment:
Some gradual improvements I can see:
1. Have each targets json as variable, i.e: `target_1`, `target_2` and then do `ONLINE_TARGET_DATABASE=[target_1, target_2]` and `OFFLINE_TARGET_DATABASE=[target_1]`, etc.
2. Construct the `*_DATABASE` in each test, by using `target_1`, `target_2` json chunks.
3. _Maybe_ have target factory of sorts for tests using `factory_boy` or similar <- this is a stretch and not sure how useful since this is only used here.

Reply:
I will implement 1 and 2. I think 3 is too much extra complexity for now, and, as you mentioned, probably not that useful at the moment.

Reply:
Refactored in 90fbe8253d3f23a6828c8e2121120f85b858bab7 - let me know what you think.

Reply:
That looks better, `*_DATABASE` are still created outside of where they'd be used, which is what I was talking about in 2.

How about:
```python
TARGET_1 = {
}  # Regular dict
TARGET_2 = {
}  # Regular dict
```

And then:

```python
mock_online_targets.json_dump.return_value = json.dumps([TARGET_1, TARGET_2])
diffs = sync_target_database.diff_objects_from_json_str(
    mock_online_targets.json_dump(), json.dumps([TARGET_1])
)
self.assertEqual(diffs, [TARGET_2])
```

Thoughts?

Reply:
I think that's a good suggestion. However, given the simplicity of the tests I think I'll leave it for now. I will make further improvements after I've got the offline mode implementation complete. Does that sound reasonable?

Reply:
It does indeed.